src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -481,19 +482,18 @@ public int hashCode() {
 
   /** Lookup the remote cache for the given {@link RemoteAction}. {@code null} if not found. */
   @Nullable
-  public RemoteActionResult lookupCache(RemoteAction action)
+  public CachedActionResult lookupCache(RemoteAction action)
```

Comment:
Please keep this method returning `RemoteActionResult`. We may add spawn-specific types there in the future.

Reply:
So do you want to put the cache name inside RemoteActionCache?
It felt weird to update that but okay... 
It was pretty straightforward to just create the RemoteActionCache in the two places that  called lookupCache.

I'll try and make this next round of changes tomorrow, but I'm wondering if it would be way more efficient for you take over the PR.

Or maybe @larsrc-google wants to help out for old-times-sake?

Reply:
> So do you want to put the cache name inside RemoteActionCache?

Yes. The design for `RemoteActionResult` instead of using `ActionResult` or `CachedActionResult` is we want a type to combine spawn-specific types and remote cache look up result which can be passed around late.

By taking over the PR, did you mean I create a folk of this PR and do the changes there?

Reply:
It wasn't clear how much back and forth there would be. i.e. RemoteActionResult has 41 usages (but mostly) from tests and I wasn't sure if I should just pass in a  CachedActionResult or the name, but I think I understand your intent now.
If not, it might be close enough for you to fork and finish.  i.e. its not clear what to do with RemoteActionResult#createFromResponse.  I'll assume cacheName being null is the right thing.